app/src/main/java/app/klimatic/ui/utils/ViewState.kt

```diff
@@ -0,0 +1,18 @@
+package app.klimatic.ui.utils
+
+import app.klimatic.ui.utils.ErrorUtils.ERROR_CODE_GENERIC
+
+sealed class ViewState<ResultType> {
+
+    data class Success<ResultType>(
+        val data: ResultType
+    ): ViewState<ResultType>()
+
+    data class Error<ResultType>(
+        val code: Int? = ERROR_CODE_GENERIC
+    ) : ViewState<ResultType>()
+
+    class StartLoading<ResultType> : ViewState<ResultType>()
```

Comment:
Shouldn't be **ShowLoading** ? If below method is **HideLoading**
Or rename below method to **StopLoading**

Reply:
Will correct in next PR